app/Http/Controllers/Auth/AuthController.php

```diff
@@ -179,8 +185,55 @@ public function postOauth(Request $request)
         return redirect($this->redirectPath());
     }
 
+    /**
+     * Show form to collect new username.
+     * 
+     * @return Response
+     */
     public function getOauth()
     {
         return view('/errors/oauthname');
     }
+
+    /**
+     * Send a mail to register a user.
+     * 
+     * @param  Request $request
+     * @return Response
+     */
+    public function sendMail(Request $request)
+    {
+        $validator = $this->validator($request->all());
+
+        if ($validator->fails()) {
+            $this->throwValidationException(
+                $request, $validator
+            );
+        }
+
+        $transport = \Swift_SmtpTransport::newInstance(env('MAIL_HOST'), env('MAIL_PORT'))
+                        ->setUsername(env('MAIL_USERNAME'))
+                        ->setPassword(env('MAIL_PASSWORD'));
+
+        $mailer = \Swift_Mailer::newInstance($transport);
+
+        $_url = $request->url();
+        $url = substr($_url, 0, stripos($_url, '/auth/register'))."?_token={$request->_token}";
+
+        $message = \Swift_Message::newInstance('Confirm your email address.')
+                    ->setFrom([env('MAIL_USERNAME') => 'Team Pibbble'])
+                    ->setTo([$request->email => $request->username])
+                    ->setBody("Dear {$request->username},<br><br>
+                        Thank you for registering with us. Confirm your email with the link below.<br>
+                        <a href={$url}>CONFIRM EMAIL</a>", 'text/html');
+
+        $mailer->send($message);
+
+        $request->session()->put('_token', $request->_token);
+        $request->session()->put('username', $request->username);
+        $request->session()->put('email', $request->email);
```

Comment:
I used the data stored in the session variable later in the ValidateEmail
middleware.

An HTTP address does not maintain state as a regular desktop or mobile app
does.
The session variable enables us to store data particular to one user
session.
It is used to maintain state across several pages of a web application for
a particular user session.
By default, a session ends when the user closes the browser. It could also
end after a period of time has elapsed.

On Mon, Nov 23, 2015 at 12:30 PM, Prosper Otemuyiwa <
notifications@github.com> wrote:

> In app/Http/Controllers/Auth/AuthController.php
> <https://github.com/andela/pibbble/pull/141#discussion_r45593476>:
>
> > +
> > +        $_url = $request->url();
> > +        $url = substr($_url, 0, stripos($_url, '/auth/register'))."?_token={$request->_token}";
> > +
> > +        $message = \Swift_Message::newInstance('Confirm your email address.')
> > +                    ->setFrom([env('MAIL_USERNAME') => 'Team Pibbble'])
> > +                    ->setTo([$request->email => $request->username])
> > +                    ->setBody("Dear {$request->username},<br><br>
> > +                        Thank you for registering with us. Confirm your email with the link below.<br>
> > +                        <a href={$url}>CONFIRM EMAIL</a>", 'text/html');
> > +
> > +        $mailer->send($message);
> > +
> > +        $request->session()->put('_token', $request->_token);
> > +        $request->session()->put('username', $request->username);
> > +        $request->session()->put('email', $request->email);
>
> Nope..I still don't know why you used it.. Please explain in clear terms
> why you needed sessions here
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/andela/pibbble/pull/141/files#r45593476>.
>



-- 
*Opeyemi Ogunjimi | Software Developer | Andela*
(c) +234-703-936-2475 | Andela.com <http://andela.com/>